Mathlib/Data/Matrix/Basic.lean

```diff
@@ -1209,6 +1210,87 @@ def diagonalAlgHom : (n → α) →ₐ[R] Matrix n n α :=
 
 end Algebra
 
+section AddHom
+
+variable [Add α]
+
+variable (R α) in
+/-- Extracting entries from a matrix as an additive homomorphism.  -/
+@[simps]
+def elemAddHom (i : m) (j : n) : AddHom (Matrix m n α) α where
+  toFun M := M i j
+  map_add' _ _ := rfl
+
+-- The type ascription on the RHS is necessary for unification to succeed on the composition.
+lemma elemAddHom_eq_proj {i : m} {j : n} :
+    elemAddHom α i j = (Pi.evalAddHom _ j).comp (Pi.evalAddHom _ i : AddHom _ (n → α)) :=
+  rfl
+
+end AddHom
+
+section AddMonoidHom
+
+variable [AddZeroClass α]
+
+variable (R α) in
+/--
+Extracting entries from a matrix as an additive monoid homomorphism. Note this cannot be upgraded to
+a ring homomorphism, as it does not respect multiplication.
+-/
+@[simps]
+def elemAddMonoidHom (i : m) (j : n) : Matrix m n α →+ α where
+  toFun M := M i j
+  map_add' _ _ := rfl
+  map_zero' := rfl
+
+-- The type ascription on the RHS is necessary for unification to succeed on the composition.
+lemma elemAddMonoidHom_eq {i : m} {j : n} :
+    elemAddMonoidHom α i j =
+      (Pi.evalAddMonoidHom _ j).comp (Pi.evalAddMonoidHom _ i : _ →+ (n → α)) :=
```

Comment:
This naming is fine also (i.e., not specifying the RHS), but let's be consistent throughout for parallelism.